src/lib/chawathe98_matcher.rs

```diff
@@ -1254,20 +1214,19 @@ fn build_colour_map() -> HashMap<ActionType, term::color::Color> {
     map.insert(ActionType::INSERT, term::color::BRIGHT_GREEN);
     map.insert(ActionType::MOVE, term::color::BRIGHT_BLUE);
     map.insert(ActionType::UPDATE, term::color::BRIGHT_YELLOW);
-    map.insert(ActionType::COPY,term::color::BRIGHT_MAGENTA);
-    map.insert(ActionType::GLUE,term::color::BRIGHT_CYAN);
+    map.insert(ActionType::COPY, term::color::BRIGHT_MAGENTA);
+    map.insert(ActionType::GLUE, term::color::BRIGHT_CYAN);
     map
 }
-/*
-            ********** END  - Printing in terminal ************
-
-*/
-
+//            ********** END  - Printing in terminal ************
 
 /// Gets the corresponding From Node Index using the To Node Index.
-pub fn get_index_from_node(map : &HashMap<usize,usize>,to_node:usize) -> usize{
-    let mut output:usize = if map.get(&to_node).is_some() { return map[&to_node]
-    } else { return 99999 };
+pub fn get_index_from_node(map: &HashMap<usize, usize>, to_node: usize) -> usize {
+    let mut output: usize = if map.get(&to_node).is_some() {
+        return map[&to_node];
+    } else {
+        return 99999;
```

Comment:
What's this funny value mean?

Reply:
Fixed. It was me giving it a worst case value.

Reply:
What was the fix?

Reply:
having it assign the max value of usize "<usize>::max_value()"

Reply:
What happens if that value "escapes" into the wider code? Is it still safe or should it panic or ...?

Reply:
I check if the value is within the bounds of 'from_arena', if it is not then edit script operation will be not applied.

Reply:
I'm not keen on this magic value, because a) it's undocumented (admittedly that's easily fixed) b) what happens if another not-magic-but-looks-like-it escapes? Let's say `from_arena` is of length 3, but we accidentally have an off-by-1 error somewhere that releases `3` into the wild. That will now be treated the same as `usize::max_value()` I think, which will cause havoc.

Would it be possible to have this method either a) have its return value checked immediately b) return `Option<usize>` to indicate when failure has occurred?